azkaban-exec-server/src/main/java/azkaban/container/ContainerServlet.java

```diff
@@ -0,0 +1,187 @@
+    throws IOException {
+    handleRequest(req, resp);
+  }
+
+  public void handleRequest(final HttpServletRequest req, final HttpServletResponse resp)
+    throws IOException {
+    final HashMap<String, Object> respMap = new HashMap<>();
+    try {
+      if (!hasParam(req, ConnectorParams.ACTION_PARAM)) {
+        logger.error("Parameter action not set");
+        respMap.put("error", "Parameter action not set");
+      } else if (!hasParam(req, ConnectorParams.EXECID_PARAM)) {
+        logger.error("Parameter execId not provided");
+        respMap.put(ConnectorParams.RESPONSE_ERROR, "Parameter execId not provided");
+      } else {
+        final String action = getParam(req, ConnectorParams.ACTION_PARAM);
+        if (!(action.equals(ConnectorParams.CANCEL_ACTION) ||
+              action.equals(ConnectorParams.LOG_ACTION)) ||
+              action.equals(ConnectorParams.METADATA_ACTION)) {
+          // Only the above 3 actions are supported for Containerized ecosystem
+          respMap.put(ConnectorParams.RESPONSE_ERROR, "Unsupported action type: " + action);
+        } else {
+          final int execid = Integer.parseInt(getParam(req, ConnectorParams.EXECID_PARAM));
+          final String user = getParam(req, ConnectorParams.USER_PARAM, null);
+
+          logger.info("User " + user + " has called action " + action + " on " + execid);
+
+          if (action.equals(ConnectorParams.CANCEL_ACTION)) {
+            handleAjaxCancel(respMap, execid, user);
+          } else if (action.equals(ConnectorParams.METADATA_ACTION)) {
+            handleFetchMetaDataEvent(execid, req, resp, respMap);
+          } else {
+            // action == LOG_ACTION
+            handleFetchLogEvent(execid, req, resp, respMap);
+          }
+        }
+      }
+    } catch (Exception e) {
+      logger.error(e.getMessage(), e);
+      respMap.put(ConnectorParams.RESPONSE_ERROR, e.getMessage());
+    }
+    writeJSON(resp, respMap);
+    resp.flushBuffer();
+  }
+
+  private void handleAjaxCancel(final Map<String, Object> respMap, final int execid,
+      final String user) {
+    if (user == null) {
+      respMap.put(ConnectorParams.RESPONSE_ERROR, "user has not been set");
+      return;
+    }
+
+    try {
+      this.flowContainer.cancelFlow(execid, user);
+      respMap.put(ConnectorParams.STATUS_PARAM, ConnectorParams.RESPONSE_SUCCESS);
+    } catch (final ExecutorManagerException e) {
+      logger.error(e.getMessage(), e);
+      respMap.put(ConnectorParams.RESPONSE_ERROR, e.getMessage());
+    }
+  }
+
+  private void handleFetchLogEvent(final int execId, final HttpServletRequest req,
+      final HttpServletResponse resp, final Map<String, Object> respMap)
+      throws ServletException {
+    final String type = getParam(req, "type");
+    final int startByte = getIntParam(req, "offset");
+    final int length = getIntParam(req, "length");
+
+    resp.setContentType("text/plain");
```

Comment:
As text/plain and utf-8 are used at many places, it is good to have constant for it.